### DISCUSSION TITLE
Support for GeoParquet

### DISCUSSION THREAD

**Discussion by bfgdigital** [duckdb/duckdb-spatial#79] [Q&A] [answered]
Hi team, great work on this, its a really promising solution to improving the performance of some of our big queries.

I'm trying to figure out the best workflow to incorporate Spatial when regular parquet/delta/etc files don't support geo types?
The premise I'm working under is that efficient data formats combined with efficient memory formats enable significant performance improvements. 

The area i'm having difficulty is marrying the two. 
I've looked into reading a db with geopandas and exporting it as nicely partitioned geoparquet files as one-off batch jobs, which I can then query efficiency as need be.

We've got 200gb+ of data is in PostgreSQL using PostGIS and getting it into duckdb means writing the data out of PostgreSQL into a format that can then be read back in to memory. 

Our experiments so far have shown we can write a 15gb 45M row table to standard partitioned parquet with an 8x memory saving (low cardinality data). That suddenly makes in-memory processing much more achievable, but getting the data into duckdb has been the difficulty. 

Could be that I've overlooked something, but at the moment I'm struggling with getting the geo data with the geom types preserved into duckdb under the conditions in which ducdb shines, where the data is small enough to read into memory.

I'd love to hear your thoughts and maybe get some suggestions on getting our existing setup into a more workable state.

![Screenshot 2023-5](https://github.com/duckdblabs/duckdb_spatial/assets/47546089/e3c1d859-3559-4c32-8c61-8e14835b61f5)

**Comment by Maxxen** [comment, marked as answer]
Hi!
Improving the geo support in our parquet reader/writer is something were looking into, but for now you should be able to work around it. Since geoparquet stores geometry as "WKB" in binary columns with some special metadata (which we dont read yet), you can just convert the blob data to duckdb geometry using the ST_GeomFromWKB(BLOB) function.

Since you mentioned postgis, you could also try to use the duckdb postgres scanner to scan postgres tables directly, using the ST_GeomAsWKB function on the  postgis side and then ST_GeomFromWKB on the duckdb side to convert the geometry columns, skipping the parquet step entirely.

**Reply by bfgdigital** [reply]
That's really helpful, thanks Maxxen.
I'm going to give these a try.
Appreciate the fast response.

**Reply by Maxxen** [reply]
No problem! We had a bug were ST_GeomFromWKB did not accept a blob, even though it is fixed I don't think the binary we distribute when downloading from the CLI (the one you get when you do `INSTALL 'spatial'` has this fix yet. If you're still on DuckDB 0.7.1 you can try to grab some artefacts from the CI runs here on GitHub, like (https://github.com/duckdblabs/duckdb_spatial/actions/runs/4873974966) although if you want all the fixes I'd recommend one of the later builds that try to stay compatible with the DuckDB master branch. We've had some breaking changes between  0.7.1->0.8.0 but we're looking to release a stable 0.8.0 this week, at which point I'll issue a new release for the spatial extension as well.

**Reply by mtravis** [reply]
@bfgdigital was just wondering if you ever tested the ability to scan postgres tables in duckdb and how you found it in terms of the spedd it took to scan?
